libcxx/src/CMakeLists.txt

```diff
@@ -173,6 +173,12 @@ if (APPLE AND LLVM_USE_SANITIZER)
   endif()
 endif()
 
+include(CheckCXXSymbolExists)
+check_cxx_symbol_exists("copy_file_range" "unistd.h" LIBCXX_USE_COPY_FILE_RANGE)
+if(LIBCXX_USE_COPY_FILE_RANGE)
+  list(APPEND LIBCXX_COMPILE_FLAGS "-D_LIBCPP_FILESYSTEM_USE_COPY_FILE_RANGE")
+endif()
```

Comment:
What Linux versions don't provide `copy_file_range`? I'd rather assume support from the OS or test for something general like `#ifdef __linux__` in `operations.cpp` than add a CMake check for a single OS feature like this.

Reply:
since linux 4.5, according to the man page. RHEL 8 is at linux 4.18, for comparison. I don't think any still supported distro uses < 4.5
On FreeBSD, it has been added in FreeBSD 13, and FreeBSD 12.x went EoL on 31. December 2023

So yeah, if you want I'll just drop this and do an `#ifdef __linux__ || __FreeBSD__`